src/QuadForm/Lattices.jl

```diff
@@ -966,6 +966,46 @@ function restrict_scalars(L::AbsLat)
   return ZLat(Vabs, Mabs)
 end
 
+@doc Markdown.doc"""
+    restrict_scalars_with_map(L) -> 
+       Tuple{ZLat, Hecke.SpaceRes}}
+
+Given a hermitian lattice `L`, return the $\mathbb Z$-lattice obtained by 
+restriction of scalars, together with the map `f` for extending scalars back.
+"""
+function restrict_scalars_with_map(L)
+  V = ambient_space(L)
+  Vabs, f = restrict_scalars(V, FlintQQ)
+  Babs = absolute_basis(L)
+  Mabs = zero_matrix(FlintQQ, length(Babs), rank(Vabs))
+  for i in 1:length(Babs)
+    v = f\(Babs[i])
+    for j in 1:length(v)
+      Mabs[i, j] = v[j]
+    end
+  end
+  return ZLat(Vabs, Mabs), f
+end
+
+@doc Markdown.doc"""
+    restrict_scalars_with_respect_to_map(L, f) -> ZLat
```

Comment:
```suggestion
    restrict_scalars(L::AbsLat, f::SpaceRes) -> ZLat
```
After discussion with reviewers, it seems that the name is a bit too much for what it is, so we shall leave it as `restrict_scalars`, like the first method, but this one would have two arguments.